staging/src/k8s.io/kubelet/config/v1beta1/types.go

```diff
@@ -803,6 +803,15 @@ type KubeletConfiguration struct {
 	// Default: true
 	// +optional
 	EnableSystemLogHandler *bool `json:"enableSystemLogHandler,omitempty"`
+	// ShutdownGracePeriod specifies the total duration that the node should delay the shutdown and total grace period for pod termination during a node shutdown.
+	// Default: "30s"
+	// +optional
+	ShutdownGracePeriod metav1.Duration `json:"shutdownGracePeriod,omitempty"`
+	// ShutdownGracePeriodCriticalPods specifies the duration used to terminate critical pods during a node shutdown. This should be less than ShutdownGracePeriod.
+	// For example, if ShutdownGracePeriod=30s, and ShutdownGracePeriodCriticalPods=10s, during a node shutdown the first 20 seconds would be reserved for gracefully terminating normal pods, and the last 10 seconds would be reserved for terminating critical pods.
+	// Default: "10s"
+	// +optional
+	ShutdownGracePeriodCriticalPods metav1.Duration `json:"shutdownGracePeriodCriticalPods,omitempty"`
```

Comment:
One small thought I have while reviewing your PR: We might want to differentiate system-cluster-critical from system-node-critical. We might want to gracefully terminate the normal pods and system-cluster-critical pods first, then terminate system-node-critical pods. But I am ok with the current approach for now.

Reply:
Thanks Dawn for taking a look. Good point, both categories of critical pods are included in this duration. 

I agree this something we can re-evaluate in beta of the feature and determine if it's worth splitting these two categories out during shutdown.